### PR TITLE
Remove unnecessary dependencies from R package

### PR DESCRIPTION
**Pull request by smurching**
## What changes are proposed in this pull request?
 
Removes `dplyr`, `packrat`, `pryr`  dependencies from MLflow R package, to help minimize the likelihood of dependency conflicts with existing user environments.
 
## How is this patch tested?
 
Existing unit tests (in particular, unit tests already exist for `mlflow_search_runs` and `mlflow_list_run_infos`, whose implementations had to change slightly to accommodate the removal of `dplyr`).
 
### Is this a user-facing change? 

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by kevinykuo** [comment]
packrat and pryr can probably go, too, while you're at it ;)

File: mlflow/R/mlflow/R/tracking-runs.R [hunk 1]
```diff
@@ -275,8 +275,9 @@ mlflow_search_runs <- function(filter = NULL,
     run_view_type = run_view_type
   ))
 
-  response$run %>%
-    purrr::map_df(parse_run)
+  runs_list <- response$run %>%
+    purrr::map(parse_run)
+  do.call("rbind", runs_list) %||% data.frame()
```
Comment (tomasatdatabricks) [inline mlflow/R/mlflow/R/tracking-runs.R]: Nice. 

I wonder if there is any difference in performance. Maybe we should put a comment to add the dplyr back after 1.0 and update minimum rlang version to 0.3.4?

File: mlflow/R/mlflow/R/tracking-utils.R [hunk 2]
```diff
@@ -140,9 +140,17 @@ parse_run <- function(r) {
   new_mlflow_run(info)
 }
 
+fill_missing_run_cols <- function(r) {
+  expected_names <- c("run_uuid", "experiment_id", "user_id", "status", "start_time",
```
Comment (tomasatdatabricks) [inline mlflow/R/mlflow/R/tracking-utils.R]: I am not sure if this is necessary. If we want to keep it we should have a test to make sure the hardcoded list is up to date.

**Review by tomasatdatabricks — COMMENTED** [review]
Looks great. I left small comments.

File: mlflow/R/mlflow/R/tracking-utils.R (hunk 2 above)
Comment (smurching) [reply mlflow/R/mlflow/R/tracking-utils.R]: (from offline discussion) it's necessary to make sure the run lists all have the same set of names, will add the test!

File: mlflow/R/mlflow/R/tracking-runs.R (hunk 1 above)
Comment (smurching) [reply mlflow/R/mlflow/R/tracking-runs.R]: Yeah my guess is that this approach is probably slower. Will add the comment

File: mlflow/R/mlflow/R/tracking-utils.R (hunk 2 above)
Comment (tomasatdatabricks) [reply mlflow/R/mlflow/R/tracking-utils.R]: Thanks!

**Review by tomasatdatabricks — APPROVED** [review]
lgtm!

**Pull request merged**